src/runtime/recipe/internal/handle.ts

```diff
@@ -256,10 +256,13 @@ export class Handle implements Comparable<Handle>, PublicHandle {
     return this._capabilities;
   }
 
-  private hasHardRef(connection: HandleConnection): boolean {
-    if (connection.type && connection.type.getEntitySchema()) {
-      for (const field of Object.values(connection.type.getEntitySchema().fields)) {
-        if (field.annotations.some(a => a.name === 'hardRef')) {
+  private hasHardRef(schema: Schema): boolean {
+    for (const field of Object.values(schema.fields)) {
+      if (field.annotations.some(a => a.name === 'hardRef')) {
+        return true;
+      }
+      if (field.isNested || field.isOrderedList) {
```

Comment:
I think you could remove this if statement at all - `if (field.isNested || field.isOrderedList) {`. 
if `getEntityType()` is not null, then we should look at its schema (so just this one - `if (field.getEntityType() && this.hasHardRef(field.getEntityType().getEntitySchema())) {` is enough).

Reply:
I don't want to follow references. If entity A has a reference to entity B, and entity B has an hard reference, entity A does not need the capability (since it has only a normal reference to entity B).
Added a test case for this. So leaving as it is?